app/uk/gov/hmrc/gform/models/FormModelBuilder.scala

```diff
@@ -65,14 +65,17 @@ object FormModelBuilder {
     formModel: FormModel[T],
     phase: Option[FormPhase]
   ): Boolean = {
+
+    def booleanExprResolver = BooleanExprResolver(loop)
```

Comment:
why is this a def? 

Also, seems like there is some duplication with this evalIncludeIf and the one in Recalculation (evalIncludeIfPure).

Reply:
This needs to be def, otherwise `loop` would be a forward reference.